drivers/python/rethinkdb/net.py

```diff
@@ -243,6 +251,20 @@ def __init__(self, parent, timeout):
                 socket.create_connection((self.host, self.port), timeout)
             self._socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
 
+            if self._use_ssl:
+                ssl_context = self._get_ssl_context(self._ca_certs)
+                try:
+                    self._socket = ssl_context.wrap_socket(self._socket,
+                                                           server_hostname=self.host)
+                except IOError as exc:
+                    self._socket.close()
+                    raise RqlDriverError("SSL handshake failed: %s" % (str(exc),))
+                try:
+                    match_hostname(self._socket.getpeercert(), hostname=self.host)
```

Comment:
Tornado doesn't use this code path because it uses `tornado.iostream` instead.  This will need to be added to `net_tornado.py`:`ConnectionInstance` for SSL to work there.  Should be fairly easy using [`tornado.iostream.SSLIOStream`](http://tornado.readthedocs.org/en/latest/iostream.html#tornado.iostream.SSLIOStream).

For that class, it says that when connecting (which we do asynchronously), the future will be fulfilled when both the connection and handshake have completed.